### DISCUSSION TITLE
Video 314: Running FoodVision Mini on Hugging Face Spaces - Build Error

### DISCUSSION THREAD

**Discussion by sankar82** [mrdbourke/pytorch-deep-learning#673] [Q&A]
I am trying to deploy the FoodVision Mini app to Hugging Face Spaces but I am encountering "Build Error" repeatedly. I removed share=True, I also used the same versions for torch, torchvision and gradio as in the video in the requirements.txt file. 

This is the error message I am getting: I have also attached a screenshot of the error log. 

`× python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-a6t9lqpp/uvicorn_7105b4d0ad5a4749adae006c4b6b5908/setup.py", line 41, in <module>
          long_description=get_long_description(),
        File "/tmp/pip-install-a6t9lqpp/uvicorn_7105b4d0ad5a4749adae006c4b6b5908/setup.py", line 23, in get_long_description
          return open('README.md', 'r').read()
      FileNotFoundError: [Errno 2] No such file or directory: 'README.md'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

[notice] A new release of pip available: 22.3.1 -> 23.2.1
[notice] To update, run: python -m pip install --upgrade pip

--> ERROR: process "/bin/sh -c pip install --no-cache-dir -r requirements.txt" did not complete successfully: exit code: 1`

Appreciate any help or pointers on how to resolve this. Thanks a lot. 
<img width="811" alt="Hugging Face error" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/41609359/e74fc934-bd87-4782-97a4-ef8775b1261d">

**Comment by mrdbourke** [comment]
hey there, thank you for posting this

I'm having this issue too with FoodVision Big and FoodVision Mini, see here: https://github.com/mrdbourke/pytorch-deep-learning/issues/664 

Currently trying to figure it out, I'll post back here when I figure out what's going on!

**Reply by sankar82** [reply]
Aah okay, thanks. Great. I am on verge of completing the course. This was the only thing pending! Thanks a lot.

**Reply by Humaan10** [reply]
@mrdbourke I am also experiencing the same issue. I will really appreciate if you provide us with information as soon as possible. Thanks

**Reply by skycomv2** [reply]
I had a different build error than you on hugging faces, but for me I solved it by changing the requirements.txt to just:

torch
torchvision
gradio

removing the == version number 

Then my hugging face ran correctly. Hope this helps someone else in the future.

**Reply by bofcarbon1** [reply]
Do you have a generated README.md file at the root of your space?
Hugging Face will create one and may expect it to be there.

**Comment by Rayden98** [comment]
it's in the name. must be "requirements.txt" .....................the wrong was "requirements.text"

**Reply by bofcarbon1** [reply]
The 'requirements.txt' file is very useful. The more packages you have the more chances that over time future clones of the app might produce problematic pairings by just taking the latest version by default. You can actually have all the packages you need installed from the requirements.txt file.
